src/sid/shared.py

```diff
@@ -28,6 +28,9 @@ def factorize_assortative_variables(states, assort_by, is_recurrent):
     variables. If there are no assortative variables, a single group is assigned to all
     states.
 
+    The unique values of the ``assort_by`` variables should be sorted to maintain the
```

Comment:
I think this is not necessary anymore. Since you are using ` groupby.indices` to create the group indexers the order of the factorized variables is irrelevant. At least it looked like that in a few tests I made.

Reply:
Done